src/services/db/GitFileSystemService.ts

```diff
@@ -955,6 +956,33 @@ export default class GitFileSystemService {
           const path = directoryPath === "" ? name : `${directoryPath}/${name}`
           const type = isDirectory ? "dir" : "file"
 
+          if (includeSha) {
+            return this.getGitBlobHash(repoName, path, isStaging)
+              .orElse(() => okAsync(""))
+              .andThen((sha) =>
+                ResultAsync.combine([
+                  okAsync(sha),
+                  this.getFilePathStats(repoName, path, isStaging),
+                ])
+              )
+              .andThen((shaAndStats) => {
+                const [sha, stats] = shaAndStats
+                return this.getFilePathStats(repoName, path, isStaging).andThen(
```

Comment:
can probably remove the call to `this.getFilePathStats` cos we alr have it from prev step

Reply:
fixed